### PR TITLE
[IMP] Inventory: Reordering rules 001 update

### PR DESCRIPTION
**Pull request by hojo-odoo**
This PR updates the Reordering rules doc to add a section detailing how to configure and use 0/0/1 reordering rules.

### CODE REVIEW

**Comment by hojo-odoo** [comment]
Hey @Felicious,

This PR is ready for review.

Thanks,
John

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 1]
```diff
@@ -93,6 +95,64 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+At the top of the product's page, click the :icon:`fa-refresh` (Reordering Rules) smart button to
+open the reordering rules page for the product. On the page, click :guilabel:`New` to begin
+configuring a new reordering rule.
+
+In the :guilabel:`Route` field of the new reordering rule, select the route the rule should use to
+replenish the item. For example, if the product should be purchased from a vendor, select the
+:guilabel:`Buy` route.
+
+In the :guilabel:`Min Quantity` field and :guilabel:`Max Quantity` field, leave the values set to
+`0.00`. In the :guilabel:`To Order` field, enter a value of `1.00`.
```
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:133]: To my knowledge, we don't manually enter `1 ` in the To Order field. We'd create a sales order so that forecasted quantity is -1, and Odoo automatically increments the To Order value to 1.
```suggestion
`0.00`. With this set up, once an order is placed, the reordering rule is triggered to bring the negative stock value back to 0.
```
Also, feel free to reword this as you see fit! :D

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 2]
```diff
@@ -93,6 +95,64 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+With the reordering rule configured using these values, each time an |SO| causes the forecasted
+quantity of the product to fall below the :guilabel:`Min Quantity` of `0.00`, the selected
+:guilabel:`Route` is used to replenish the product in one-unit increments, back up to the
+:guilabel:`Max Quantity` of `0.00`.
+
+.. example::
+   A picture frame is configured with a 0/0/1 reordering rule that uses the *Buy* route. Zero units
+   of the picture frame are kept on-hand at any given time.
+
+   A purchase order (PO) is confirmed for one unit of the picture frame, which causes the forecasted
```
Comment (Felicious) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:148]:
```suggestion
   A sales order is confirmed for one unit of the picture frame, which causes the forecasted
```

**Review by Felicious — APPROVED** [review]
Hi @hojo-odoo ! 

Great job with this PR! It'll be really helpful for people learning about 0,0 (: 

I just had some suggestions to really elaborate on the functionality of the feature. Let me know if there's anything I can help with or clarify!

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst (hunk 1 above)
Comment (hojo-odoo) [reply content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:133]: I just created a new reordering rule to confirm, and the default To Order value is 0.

**Comment by hojo-odoo** [comment]
Hey @ksc-odoo,

This PR is ready for final review.

Thanks,
John

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 3]
```diff
@@ -93,6 +95,66 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+
+The *0/0/1* reordering rule is a specialty rule used to replenish a product that is not kept
+on-hand, each time a sales order (SO) is confirmed for that product.
+
+.. important::
+   The 0/0/1 reordering rule is similar to the *Replenish on Order (MTO)* route, in that both
+   workflows are used to replenish a product upon confirmation of an |SO|.
+
+   The main difference between the two methods is that the Replenish on Order route automatically
+   reserves the product for the |SO| that caused it to be replenished. This means the product cannot
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:109]:
```suggestion
   reserves the product for the |SO| that caused it to be replenished. This means the product **cannot**
```
SUGGESTION: just for added emphasis

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 4]
```diff
@@ -93,6 +95,66 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+   linked to the original |SO| by a smart button at the top of the order. When using the 0/0/1
+   reordering rule, a replenishment order is created, but is not linked to the original |SO|.
+
+   See the :doc:`Replenish on Order (MTO) <mto>` documentation for a full overview of the MTO route.
+
+To create a 0/0/1 reordering rule, navigate to :menuselection:`Inventory app --> Products -->
+Products`, and select a product.
+
+At the top of the product's page, click the :icon:`fa-refresh` :guilabel:`Reordering Rules` smart
+button to open the :guilabel:`Reordering Rules` page for the product. On the page, click
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:125]:
```suggestion
button to open the :guilabel:`Reordering Rules` page for the product. On the resulting page, click
```

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 5]
```diff
@@ -93,6 +95,66 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+   A picture frame is configured with a 0/0/1 reordering rule that uses the *Buy* route. Zero units
+   of the picture frame are kept on-hand at any given time.
+
+   An |SO| is confirmed for one unit of the picture frame, which causes the forecasted quantity to
+   drop to `-1.00`. This triggers the reordering rule, which automatically creates a |PO| for one
+   unit of the picture frame.
+
+   Once the product is received from the vendor, the forecasted quantity of the picture frame returns
+   to `0.00`. There is now one picture frame on-hand, but it is not reserved for the SO which
+   triggered its purchase. It can be used to fulfill that SO, or reserved for a different order.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:156]: just a suggestion, do you want to consider using the abbreviate |SO| formatting for these last two mentions, or just leave them as they are. Totally fine either way -- completely up to you -- just thought it was worth mentioning for your consideration.

**Review by ksc-odoo — APPROVED** [review]
Hey @hojo-odoo -- just finished my Final Review. Nicely done. Only a small handful of suggestions/comments that require your attention. Approving now 👍 Once you implement the necessary adjustments, feel free to tag this for tech review. Thanks! 🙏

**Comment by hojo-odoo** [comment]
Hey @samueljlieber,

This PR is ready for technical review.

Thanks,
John

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 6]
```diff
@@ -93,6 +95,66 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+---------------------
+
+The *0/0/1* reordering rule is a specialty rule used to replenish a product that is not kept
+on-hand, each time a sales order (SO) is confirmed for that product.
+
+.. important::
+   The 0/0/1 reordering rule is similar to the *Replenish on Order (MTO)* route, in that both
+   workflows are used to replenish a product upon confirmation of an |SO|.
+
+   The main difference between the two methods is that the Replenish on Order route automatically
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:108]: Italicize the route's name since you are referencing it exactly :)
```suggestion
   The main difference between the two methods is that the *Replenish on Order* route automatically
```

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 7]
```diff
@@ -93,6 +95,66 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+   reserves the product for the |SO| that caused it to be replenished. This means the product
+   **cannot** be used for a different |SO|.
+
+   The 0/0/1 reordering rule does not have this limitation. A product replenished using the rule is
+   not reserved for any specific |SO|, and can be used as needed.
+
+   Another key difference is that replenishment orders created by the Replenish on Order route are
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:115]: Here too please
```suggestion
   Another key difference is that replenishment orders created by the *Replenish on Order* route are
```

File: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst [hunk 8]
```diff
@@ -93,6 +95,66 @@ For advanced usage of reordering rules, learn about the following reordering rul
...
+   A picture frame is configured with a 0/0/1 reordering rule that uses the *Buy* route. Zero units
+   of the picture frame are kept on-hand at any given time.
+
+   An |SO| is confirmed for one unit of the picture frame, which causes the forecasted quantity to
+   drop to `-1.00`. This triggers the reordering rule, which automatically creates a |PO| for one
+   unit of the picture frame.
+
+   Once the product is received from the vendor, the forecasted quantity of the picture frame returns
+   to `0.00`. There is now one picture frame on-hand, but it is not reserved for the |SO| which
+   triggered its purchase. It can be used to fulfill that |SO|, or reserved for a different order.
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst:156]: Line too long (over 100 chars) :)
```suggestion
   Once the product is received from the vendor, the forecasted quantity of the picture frame
   returns to `0.00`. There is now one picture frame on-hand, but it is not reserved for the |SO|
   which triggered its purchase. It can be used to fulfill that |SO|, or reserved for a different
   order.
```

**Review by samueljlieber — APPROVED** [review]
Hi @hojo-odoo! Nice work on this new section in the Reordering Rules documentation on 0/0/1 RR. Your improvements look great to me, and I think you did a fantastic job at explaining this specialty rule. Nice work! Approving with a couple super small fixed. Thanks!
...
@robodoo delegate=hojo-odoo
